src/tr2/game/objects/creatures/spider.c

```diff
@@ -0,0 +1,30 @@
+#include "game/objects/creatures/spider.h"
+
+#include "game/creature.h"
+#include "game/objects/common.h"
+#include "global/const.h"
+#include "global/funcs.h"
+#include "global/vars.h"
+
+#define SPIDER_HITPOINTS 5
+#define SPIDER_RADIUS (WALL_L / 10) // = 102
+
+void Spider_Setup(void)
+{
+    OBJECT *const obj = &g_Objects[O_SPIDER];
+    if (!obj->loaded) {
+        return;
+    }
+
+    obj->control = Spider_Control;
+    obj->collision = Creature_Collision;
+
+    obj->hit_points = SPIDER_HITPOINTS;
+    obj->radius = SPIDER_RADIUS;
+    obj->shadow_size = UNIT_SHADOW / 2;
+
+    obj->save_anim = 1;
```

Comment:
Adding `obj->intelligent = 1;` fixes the spider crash @aredfan found, although if that's not part of the decomp maybe the issue is elsewhere.